nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -58,3 +72,379 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
+            else:
+                processed_cve_info["existing"].update({cve_name: self.prep_cve_for_dlc(cve)})
+        self.logger.info(
+            "Prepared %s CVE for creation." % len(processed_cve_info["new"]),
+            extra={"object": software, "grouping": "CVE Creation"},
+        )
+
+        return processed_cve_info
+
+    def query_api(self, url: str) -> dict:
+        """Query the NIST API using an established session.
+
+        Args:
+            url (str): The API endpoint being queried.
+
+        Returns:
+            dict: Dictionary of returned results if successful.
+        """
+        # Build retry configuration; Uses Integration settings but setting sane defaults in case they are removed
+        retries = Retry(
+            total=self.integration.extra_config.get("retries", {}).get("max_attempts", 3),
+            backoff_factor=self.integration.extra_config.get("retries", {}).get("backoff", 1),
+            status_forcelist=[502, 503, 504],
+            allowed_methods=["GET"],
+        )
+
+        # Create and configure session
+        session = Session()
+        session.mount("https://", HTTPAdapter(max_retries=retries))
+        session.headers.update({"apiKey": self.nist_api_key})
+
+        try:
+            result = session.get(url)
+            result.raise_for_status()
+            return result.json()
+        except HTTPError as err:
+            code = err.response.status_code
+            self.logger.error(
+                "The NIST Service is currently unavailable. Status Code: %s. Try running the job again later.", code
+            )
+            raise
+        except json.JSONDecodeError as err:
+            self.logger.error("Invalid JSON response from NIST Service: %s", err)
+            raise
+        finally:
+            session.close()
+
+    @staticmethod
+    def convert_v2_base_score_to_severity(score: float) -> str:
+        """Uses V2 Base Score to convert to Severity Value.
+
+        Args:
+            score (float): CVSS V2 Base Score
+
+        Returns:
+            str: Severity Value ["HIGH", "MEDIUM", "LOW", "UNDEFINED"]
+        """
+        if 0.0 >= score <= 3.9:
+            return "LOW"
+        if 4.0 >= score <= 6.9:
+            return "MEDIUM"
+        if 7.0 >= score <= 10:
+            return "HIGH"
+        return "UNDEFINED"
+
+    def prep_cve_for_dlc(self, cve_json: dict) -> dict:  # pylint: disable=too-many-locals
+        """Convert CVE info into a format compatible with the DLC model.
+
+        Args:
+            cve_json (dict): JSON object containing CVE information.
+
+        Returns:
+            dict: Dictionary of CVE information formatted for DLC.
+        """
+        cve_name = cve_json["id"]
+        for desc in cve_json["descriptions"]:
+            if desc["lang"] == "en":
+                cve_description = desc["value"]
+            else:
+                cve_description = "No description provided."
```

Comment:
This will only ever set the description to the last description on the list or the fallback value. Is that out intention?

Reply:
Now using next() to iterate through and only provide description for english language.